.github/workflows/contracts.test+publish.yaml

```diff
@@ -1,8 +1,9 @@
 name: "contracts: test+publish"
 
 on:
+  workflow_dispatch:
   push:
-    branches-ignore:
+    branches:
```

Comment:
This github action file includes testing so I think it'd be okay to keep the trigger as branches-ignore. The packages will be deployed only when the package version has been updated.
Then test results can be quickly checked after bing pushed into remote branches
```
if: steps.packageJsonVersion.outputs.value != steps.npmPackageVersion.outputs.VERSION
```